WordPress/src/main/java/org/wordpress/android/ui/posts/editor/media/AddLocalMediaToPostUseCase.kt

```diff
@@ -50,6 +54,45 @@ class AddLocalMediaToPostUseCase @Inject constructor(
         editorMediaListener: EditorMediaListener,
         doUploadAfterAdding: Boolean = true,
         trackEvent: Boolean = true
+    ): Boolean {
+        var result = false
+
+        if (site.hasFreePlan) {
+            uriList.map {
+                result = if (mediaUtilsWrapper.isVideo(it) &&
+                        mediaUtilsWrapper.isAllowedUploadVideoDuration(context, it)) {
+                    processMediaUri(
+                            uriList,
+                            site,
+                            freshlyTaken,
+                            editorMediaListener,
+                            doUploadAfterAdding,
+                            trackEvent)
+                } else {
+                    editorMediaListener.showVideoDurationLimitWarning(it.path.toString())
```

Comment:
Not sure if this code is reached with a multiple selection flow (i.e. when `uriList.size > 1`), but it might be a good idea to pull the warning outside the loop to avoid multiple toasts with the same warning.

Reply:
moved it outside